test/plugin/test_out_elasticsearch_dynamic.rb

```diff
@@ -474,13 +474,33 @@ def test_adds_logstash_timestamp_when_configured
     driver.configure("logstash_format true\n")
     stub_elastic_ping
     stub_elastic
-    ts = DateTime.now
-    time = Fluent::EventTime.from_time(ts.to_time)
+    begin
+      time = Fluent::EventTime.new(Time.now.to_i, 000000000)
+    rescue
+      time = Fluent::Engine.now
+    end
     driver.run(default_tag: 'test') do
-      driver.feed(time, sample_record)
+      driver.feed(time.to_i, sample_record)
     end
     assert(index_cmds[1].has_key? '@timestamp')
-    assert_equal(index_cmds[1]['@timestamp'], ts.iso8601)
+    assert_equal(index_cmds[1]['@timestamp'], Time.at(time).iso8601(9))
+  end
+
+  def test_uses_subsecond_precision_when_configured
+    driver.configure("logstash_format true
+                      time_precision 3\n")
+    stub_elastic_ping
+    stub_elastic
+    begin
+    time = Fluent::EventTime.new(Time.now.to_i, 000000000)
+    rescue
+      time = Fluent::Engine.now
+    end
+    driver.run(default_tag: 'test') do
+      driver.feed(time.to_i, sample_record)
```

Comment:
I want to use non-integer casted value.
Because v0.14’s `#feed` can accept `Integer` and `Fluent::EventTime` value.